AppController/djinn.rb

```diff
@@ -619,19 +619,36 @@ def get_app_info_map(secret)
   #   "OK" if the relocation occurred successfully, and a String containing the
   #   reason why the relocation failed in all other cases.
   def relocate_app(appid, http_port, https_port, secret)
-    if !valid_secret?(secret)
-      return BAD_SECRET_MSG
+    return BAD_SECRET_MSG if !valid_secret?(secret)
+    Djinn.log_debug("Received relocate_app for #{appid} for " +
+        " http port #{http_port} and https port #{https_port}.")
+
+    if !my_node.is_shadow?
+      # We need to send the call to the shadow.
+      Djinn.log_debug("Sending relocate_app call for #{appid} to shadow.")
+      acc = AppControllerClient.new(get_shadow.private_ip, @@secret)
+      begin
+        return acc.relocate_app(appid, http_port, https_port)
+      rescue FailedNodeException => except
+        Djinn.log_warn("Failed to forward relocate_app call to shadow (#{get_shadow}).")
```

Comment:
I am not a fan of hardcoding the shadow role in logs. Do we need it or is the IP enough?